File: go/models/race.go
package models

import "time"

type Race struct {
	Season string `json:"season"`
	Name string `json:"name"`
	DateTime time.Time `json:"time"`
	NbLaps *int8 `json:"nbLaps"`
	Laps []*Lap `json:"laps"`
}

type Lap struct {
	Number int8 `json:"number"`
	Timings []*Timing `json:"timings"`
}

type Timing struct {
	DriverId string `json:"driverId"`
	Position int8 `json:"position"`
	Time float32 `json:"time"`
}

File: vue/src/models/race.ts
export type RaceSelection = {
  season: string;
  raceNumber: string;
};

type Race = {
  season: string;
  name: string;
  time: Date;
};

export type RaceResult = Race & {
  nbLaps: number;
};

export type RaceLap = Race & {
  lapNumber: number;
  timings: Timing[];
};

export type Timing = {
  driverId: string;
  position: number;
  time: number;
};
